client/src/pages/Public.js

```diff
@@ -0,0 +1,154 @@
+  'publicGames' :{
+    url: () => '/game/public-games',
+    method: 'GET',
+    contentType: 'application/json',
+    body: '',
+  },
+  "join": {
+    url: id => `/game/${id}/join`,
+    method: "PUT",
+    contentType: "application/json",
+    body: (player, socketID) => JSON.stringify({ player, socketID }),
+  },
+}
+
+const Public = ({state, setState, socket}) => {
+  const classes = useStyles();
+  const history = useHistory();
+  const [player, setPlayer] = useState('player1');
+  const [gameList, setGameList] = useState([]);
+  const [err, setErr] = useState(undefined);
+
+  const onCreateGame = async (maxPlayerNum) => {
+    const type = 'public';
+    const res = await fetch(api[type].url(), {
+      method: api[type].method,
+      headers: {
+        'Content-Type': api[type].contentType,
+        Accept: 'application/json',
+      },
+      body: api[type].body(player, socket.id, true, maxPlayerNum),
+    });
+
+    const nextState = await res.json();
+    if (res.status >= 200 && res.status < 300) {
+      setState({
+        player: player,
+        gameID: nextState.id,
+        gameState: nextState.gameState
+      });
+      history.push(`/lobby/${nextState.id}`);
+    } else {
+      setErr(nextState.error);
+    }
+  }
+
+  const onJoin = async (id) => {
+    const testName = player;
+    const type = 'join';
+    const res = await fetch(api[type].url(id), {
+      method: api[type].method,
+      headers: {
+        "Content-Type": api[type].contentType,
+        Accept: "application/json",
+      },
+      body: api[type].body(testName, socket.id),
+    });
+
+    const nextState = await res.json();
+    if (res.status >= 200 && res.status < 300) {
+      setState({
+        player: player,
+        gameID: id,
+        gameState: nextState.gameState
+      });
+      history.push(`/lobby/${id}`);
+    } else {
+      setErr(nextState.error);
+    }
+  };
+
+  const getPublicGames = async () => {
+    const type = 'publicGames';
+    const res = await fetch(api[type].url(), {
+      method: api[type].method,
+      headers: {
+        'Content-Type': api[type].contentType,
+        Accept: 'application/json',
+      },
+    });
+    const nextState = await res.json();
+    return nextState.gameList;
+  }
+
+  const onRefresh = async () => {
+    const result = await getPublicGames();
+    setGameList(result);
+  }
+
+  useEffect(() => {
+    (onRefresh)();
+  }, []);
+
+  return (
+    <Container className={classes.root}>
+      <Dialog open={err !== undefined} onClose={() => setErr(undefined)}>
+        <DialogTitle>
+          <Typography align="center">Error</Typography>
+          <IconButton
+            className={classes.close}
+            onClick={() => setErr(undefined)}>
+            <CloseIcon/>
+          </IconButton>
+        </DialogTitle>
+        <DialogContent>
+          <Typography align="center" component="h2">
+            {err}
+          </Typography>
+        </DialogContent>
+      </Dialog>
```

Comment:
The close button isn't oriented properly

![image](https://user-images.githubusercontent.com/15241001/100494456-df90a080-310f-11eb-9388-44169f3a9f13.png)

Reply:
Fixed by 4e8a377